src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Snapshot/IDependencyExtensions.cs

```diff
@@ -39,7 +39,7 @@ public static IDependencyViewModel ToViewModel(this IDependency self, ITargetedD
             return new DependencyViewModel
             {
                 Caption = self.Caption,
-                FilePath = self.Id,
+                FilePath = self.Path ?? self.Id,
```

Comment:
This is wrong and would break all graph nodes. You have to pass our unique id - thats how graph nodes could locate corresponding dependencies. If you pass only Path graph could not find parent dependency in the snapshot.  Please don't check this in. Stop by to discuss your scenario please. I don't see why one would need to read IVsHierarchy.GetCanonicalName for our dependencies and if they would why our id would not work for them.